subprojects/assemble/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/internal/teamcity/CollectTeamcityMetricsTask.kt

```diff
@@ -13,21 +12,21 @@ import org.gradle.api.tasks.TaskAction
 import org.gradle.workers.WorkerExecutor
 import javax.inject.Inject
 
-public abstract class CollectTeamcityMetricsTask @Inject constructor(
+internal abstract class CollectTeamcityMetricsTask @Inject constructor(
```

Comment:
Tasks are usually public, why do you think narrowing the access is needed here?

Reply:
The contract is not to use tasks directly.

- If someone consumes this module as a dependency in another plugin
- If someone uses this class in buildscripts

Even if this modifier doesn't forbid usage, it can be a reminder like "internal" package.

Reply:
i see, thanks for the explanation